llvm/lib/Transforms/Vectorize/VectorCombine.cpp

```diff
@@ -843,6 +843,8 @@ bool VectorCombine::scalarizeVPIntrinsic(Instruction &I) {
 
   Value *ScalarOp0 = getSplatValue(Op0);
   Value *ScalarOp1 = getSplatValue(Op1);
+  if (!ScalarOp0 || !ScalarOp1)
```

Comment:
Should we use getSplatValue at the top where we used isSplatValue? That way we fail out before we do wasted work.

Reply:
Sure. Do you know why getSplatValue fails to get the splat even though isSplatValue returns true? That sounds like a bug to me?

Reply:
getSplatValue can't create new instructions, it can only return an operand of existing instructions. isSplatValue can look at shuffles and see that it splat one of the elements of its input. To get that element as a scalar you would need to insert an extractelement instruction.